src/mds/Server.cc

```diff
@@ -3506,10 +3508,10 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
       _inode->mode |= S_ISGID;
     }
   } else {
-    _inode->gid = mdr->client_request->get_caller_gid();
+    _inode->gid = owner_gid;
```

Comment:
What will happen when the requests from an old kclient ?

Reply:
With recent pull request revision, it will take `caller_{u,g}id` values to fill `owner_{u,g}id` fields.